apps/src/aiTutor/types.ts

```diff
@@ -7,19 +7,37 @@ export type ChatCompletionMessage = {
 };
 
 export type AITutorInteraction = {
-  userId: number;
-  levelId: number;
-  scriptId: number;
-  type: TutorTypes;
+  userId?: number;
+  levelId?: number;
+  scriptId?: number;
+  type: TutorType;
+  isProjectBacked?: boolean;
   prompt: string;
   status: string;
-  aiResponse: string;
+  aiResponse?: string;
 };
 
 export type Level = {
   id: number;
   type: string;
   hasValidation: boolean;
+  isProjectBacked: boolean;
+};
+
+interface Context {
+  levelId?: number;
+  scriptId?: number;
+  isProjectBacked?: boolean;
+};
+
+export interface GeneralChatContext extends Context {
+  message: string;
+};
+
+export interface ValidationCompilationContext extends Context {
+  systemPrompt: string;
+  studentCode: string;
+  tutorType: TutorType;
```

Comment:
@molly-moen `GeneralChatContext` and `ChatContext` were related in that they have some shared fields. Is defining the interfaces with `extends` like this more clear? Is that what you were getting at?